### PR TITLE
CASMMON-200: Documentation update for BREAK/FIX: CSM 1.3 Fresh install service monitor failure on loki

### PR DESCRIPTION
**Pull request by shreni123**
# Description

CASMMON-200: Documentation update for BREAK/FIX: CSM 1.3 Fresh install failure on loki - errors on prometheus-snmp-exporter serviceMonitor params error.

## Test on loki

prometheus-snmp-exporter:
serviceMonitor:
enabled: true
params:
- name: snmp1
target: 127.0.0.1
- name: snmp2
target: 10.252.0.2
- name: snmp3
target: 10.252.0.3
- name: snmp4
target: 10

ncn-m001:~/ram # curl http://10.42.0.22:9116/snmp?target=10.254.0.3|less
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP ifAdminStatus The desired state of the interface - 1.3.6.1.2.1.2.2.1.7
### TYPE ifAdminStatus gauge
ifAdminStatus{ifAlias="",ifDescr="1/1/2",ifIndex="2",ifName="1/1/2"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/20",ifIndex="20",ifName="1/1/20"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/21",ifIndex="21",ifName="1/1/21"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/24",ifIndex="24",ifName="1/1/24"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/25",ifIndex="25",ifName="1/1/25"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/26",ifIndex="26",ifName="1/1/26"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/27",ifIndex="27",ifName="1/1/27"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/28",ifIndex="28",ifName="1/1/28"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/29",ifIndex="29",ifName="1/1/29"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/30",ifIndex="30",ifName="1/1/30"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/31",ifIndex="31",ifName="1/1/31"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/32",ifIndex="32",ifName="1/1/32"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/33",ifIndex="33",ifName="1/1/33"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/34",ifIndex="34",ifName="1/1/34"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/35",ifIndex="35",ifName="1/1/35"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/37",ifIndex="37",ifName="1/1/37"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/38",ifIndex="38",ifName="1/1/38"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/39",ifIndex="39",ifName="1/1/39"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/4",ifIndex="4",ifName="1/1/4"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/40",ifIndex="40",ifName="1/1/40"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/41",ifIndex="41",ifName="1/1/41"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/42",ifIndex="42",ifName="1/1/42"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/43",ifIndex="43",ifName="1/1/43"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/44",ifIndex="44",ifName="1/1/44"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/45",ifIndex="45",ifName="1/1/45"} 2

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [ ] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Comment by spillerc-hpe** [comment]
This PR addresses the fresh install use case however it does not cover the upgrade from CSM 1.2 to 1.3 scenario.

One possible solution would be to modify the [update-customizations.sh](https://github.com/Cray-HPE/docs-csm/blob/release/1.3/upgrade/scripts/upgrade/util/update-customizations.sh) script to read this block if it exists and update it to the new format.

**Review by rustydb — APPROVED** [review]
Looks okay, but could use a thorough review from someone in @Cray-HPE/management-network

**Comment by shreni123** [comment]
> This PR addresses the fresh install use case however it does not cover the upgrade from CSM 1.2 to 1.3 scenario.
> 
> One possible solution would be to modify the [update-customizations.sh](https://github.com/Cray-HPE/docs-csm/blob/release/1.3/upgrade/scripts/upgrade/util/update-customizations.sh) script to read this block if it exists and update it to the new format.

Hi,  I have addressed your review comment by modifying update-customizations.sh file.

File: upgrade/scripts/upgrade/util/update-customizations.sh [hunk 1]
```diff
@@ -73,7 +73,7 @@ trap 'rm -f $c' EXIT
 
 cp "$customizations" "$c"
 
-# argo/cray-nls
+# argo/cray-s
```
Comment (spillerc-hpe) [inline upgrade/scripts/upgrade/util/update-customizations.sh:76]: What is the purpose of this change?

File: upgrade/scripts/upgrade/util/update-customizations.sh [hunk 2]
```diff
@@ -103,10 +103,12 @@ if [[ -z "$(yq r "$c" "spec.network.netstaticips.nmn_ncn_storage_mons")" ]]; the
   done
   yq w -i --style=single "$c" spec.kubernetes.services.cray-sysmgmt-health.cephExporter.endpoints '{{ network.netstaticips.nmn_ncn_storage_mons }}'
 fi
-
+if [[ "$(yq r "$c" "spec.kubernetes.services.cray-sysmgmt-health.prometheus-snmp-exporter.serviceMonitor.enabled")" ]]; then
+    echo "hello"
```
Comment (spillerc-hpe) [inline upgrade/scripts/upgrade/util/update-customizations.sh:107]: Is this `echo` needed?

File: upgrade/scripts/upgrade/util/update-customizations.sh [hunk 3]
```diff
@@ -103,10 +103,12 @@ if [[ -z "$(yq r "$c" "spec.network.netstaticips.nmn_ncn_storage_mons")" ]]; the
...
+    yq d -i "$c" "spec.kubernetes.services.cray-sysmgmt-health.prometheus-snmp-exporter.serviceMonitor.params.*.module"
```
Comment (spillerc-hpe) [inline upgrade/scripts/upgrade/util/update-customizations.sh:108]: This appears to delete part of the old configuration however it does not reformat the old style target list to the new style. Would this allow the chart to deploy successfully with a working configuration?

**Review by spillerc-hpe — CHANGES_REQUESTED** [review]

File: upgrade/scripts/upgrade/util/update-customizations.sh (hunk 1 above)
Comment (shreni123) [reply upgrade/scripts/upgrade/util/update-customizations.sh]: This happened by mistake so marking it as resolved.

File: upgrade/scripts/upgrade/util/update-customizations.sh (hunk 2 above)
Comment (shreni123) [reply upgrade/scripts/upgrade/util/update-customizations.sh]: Used for testing purpose. resolved

File: upgrade/scripts/upgrade/util/update-customizations.sh (hunk 3 above)
Comment (shreni123) [reply upgrade/scripts/upgrade/util/update-customizations.sh:107]: Yes, We tested and deployed the chart successfully. Attaching the logs for the reference:

**Comment by rambabubolla** [comment]
> 



> > This PR addresses the fresh install use case however it does not cover the upgrade from CSM 1.2 to 1.3 scenario.
> > One possible solution would be to modify the [update-customizations.sh](https://github.com/Cray-HPE/docs-csm/blob/release/1.3/upgrade/scripts/upgrade/util/update-customizations.sh) script to read this block if it exists and update it to the new format.
> 
> Hi, I have addressed your review comment by modifying update-customizations.sh file.
Test on loki
      prometheus-snmp-exporter:
        serviceMonitor:
          enabled: true
          params:
          - name: snmp1
            target: 127.0.0.1
          - name: snmp2
            target: 10.252.0.2
          - name: snmp3
            target: 10.252.0.3
          - name: snmp4
            target: 10



ncn-m001:~/ram # curl http://10.42.0.22:9116/snmp?target=10.254.0.3|less
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0# HELP ifAdminStatus The desired state of the interface - 1.3.6.1.2.1.2.2.1.7
# TYPE ifAdminStatus gauge
ifAdminStatus{ifAlias="",ifDescr="1/1/2",ifIndex="2",ifName="1/1/2"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/20",ifIndex="20",ifName="1/1/20"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/21",ifIndex="21",ifName="1/1/21"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/24",ifIndex="24",ifName="1/1/24"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/25",ifIndex="25",ifName="1/1/25"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/26",ifIndex="26",ifName="1/1/26"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/27",ifIndex="27",ifName="1/1/27"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/28",ifIndex="28",ifName="1/1/28"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/29",ifIndex="29",ifName="1/1/29"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/30",ifIndex="30",ifName="1/1/30"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/31",ifIndex="31",ifName="1/1/31"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/32",ifIndex="32",ifName="1/1/32"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/33",ifIndex="33",ifName="1/1/33"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/34",ifIndex="34",ifName="1/1/34"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/35",ifIndex="35",ifName="1/1/35"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/37",ifIndex="37",ifName="1/1/37"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/38",ifIndex="38",ifName="1/1/38"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/39",ifIndex="39",ifName="1/1/39"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/4",ifIndex="4",ifName="1/1/4"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/40",ifIndex="40",ifName="1/1/40"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/41",ifIndex="41",ifName="1/1/41"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/42",ifIndex="42",ifName="1/1/42"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/43",ifIndex="43",ifName="1/1/43"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/44",ifIndex="44",ifName="1/1/44"} 2
ifAdminStatus{ifAlias="",ifDescr="1/1/45",ifIndex="45",ifName="1/1/45"} 2

**Review by spillerc-hpe — APPROVED** [review]

**Review by rambabubolla — APPROVED** [review]
